upstairs/src/lib.rs

```diff
@@ -7007,34 +6947,99 @@ impl BlockIO for Guest {
         offset: Block,
         data: Buffer,
     ) -> Result<BlockReqWaiter, CrucibleError> {
-        self.read(offset, data).await
+        if !self.is_active().await {
+            return Err(CrucibleError::UpstairsInactive);
+        }
+
+        let bs = self.query_block_size().await?;
```

Comment:
Ah, is this why the check for is_active?  Because we need to know the block size?

Reply:
Yep :)